en/lessons/basics/collections.md

```diff
@@ -3,120 +3,224 @@ version: 1.3.1
-Arity is a core part of speaking about Elixir (and Erlang) code.
-It is the number of arguments a given function takes (two, in this case).
-Arity and the given name are combined with a slash. We'll talk more about this later; this knowledge will help you understand the notation for now.
+A side note about another operator that you might come across `<>`, this is an example of typeclass abstraction in haskell, it allows us to associate specific implementations of functions (and operators) with specific types. In the case of lists `<>` is exactly `++` under the hood. The full typeclass implemetation looks like this:
+
+```haskell
+instance Semigroup [a] where
+  (<>) = (++)
+```
+
+The keywords aren't important, but it should give you an intuition for when you see `<>` which just means concatenation for a specific type!
 
-### List Subtraction
+### List Difference
 
-Support for subtraction is provided via the `--/2` operator; it's safe to subtract a missing value:
+Support for difference is provided via the `\\` operator; it's safe to subtract a missing value:
 
-```elixir
-iex> ["foo", :bar, 42] -- [42, "bar"]
-["foo", :bar]
+```haskell
+Prelude> import Data.List
+Prelude Data.List> ["foo", "bar", "baz"] \\ ["quux", "bar"]
+["foo","baz"]
 ```
 
-Be mindful of duplicate values.
-For every element on the right, the first occurrence of it gets removed from the left:
+__Note__: `import` syntax may be new to you, but this allows us to pull in libraries into the repl, and fortunately for us GHC ships with a list module in its base library!
 
-```elixir
-iex> [1,2,2,3,2,3] -- [1,2,3,2]
-[2, 3]
+Be mindful of duplicate values. For every element on the right, the first occurrence of it gets removed from the left:
+
+```haskell
+Prelude Data.List> ["foo", "bar", "baz", "bar"] \\ ["quux", "bar"]
+["foo","baz","bar"]
 ```
 
-**Note:** List subtraction uses [strict comparison](../basics/#comparison) to match the values. For example:
+List difference uses an [equality](../basics/#comparison) instance for your specific type, to match values. We can see this in the type.
 
-```elixir
-iex> [2] -- [2.0]
-[2]
-iex> [2.0] -- [2.0]
-[]
+```haskell
+Prelude Data.List> :t (\\)
+(\\) :: Eq a => [a] -> [a] -> [a]
 ```
 
+The `Eq a` on the left hand side of the constraint arrow `=>` means that all `a`'s on the right must provide an implementation of equality. This allows us to compare the elements in the list and subtract them.
+
 ### Head / Tail
 
 When using lists, it is common to work with a list's head and tail.
 The head is the list's first element, while the tail is a list containing the remaining elements.
-Elixir provides two helpful functions, `hd` and `tl`, for working with these parts:
 
-```elixir
-iex> hd [3.14, :pie, "Apple"]
-3.14
-iex> tl [3.14, :pie, "Apple"]
-[:pie, "Apple"]
+Haskell provides two helpful functions, `head` and `tail`, for working with these parts:
```

Comment:
might be worth including simple pattern matching as well.

Reply:
I think pattern matching is covered in basics and I roughly demonstrate it in the `safeHead` example. I am going to leave this comment unresolved though, I think its a worthwhile subject if pattern matching isn't covered elsewhere.